aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/RequestRetry.kt

```diff
@@ -0,0 +1,45 @@
+package com.amplifyframework.datastore.syncengine
+
+import com.amplifyframework.AmplifyException
+
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.core.SingleEmitter
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+
+class RequestRetry(private val maxExponent: Int = 8,
+                    private val jitterFactor: Int = 100,
+                    private val maxAttempts: Int = 3) {
+
+    private var numberOfAttempts = 0
+
+
+        fun <T> retry(single: Single<T>, skipExceptions: List<Class<out Throwable>?>): Single<T> {
```

Comment:
This line should be indented 4 spaces, not 8.  There's some other weird indentation in this file too, so it would be good to clean it up a bit.